### DISCUSSION TITLE
How to add one more feature in label

### DISCUSSION THREAD

**Discussion by ghost** [ultralytics/yolov5#6615] [Q&A]
YOLO has object name, centerX,centerY, witdth and height features/fields as label. How can we add one more like orientation? Can someone show which lines to modify?

**Comment by glenn-jocher** [comment]
@safaATcurrus before you do anything of this sort you'd need a labelled dataset that includes your additional features.

**Reply by ghost** [reply]
I already have. Kitti dataset has those fields. I am planning to make some changes to code but I will need some guidance. Now yolo seems to be written as there won't be any data added. 

One example is:

line 77 in augmentations.py I changed the label here so that it only takes the first 5 rows. 

boxes = labels[:, 1:5].astype(int)

I will make changes now in my own copy so that it won't give any error for additional data field in the label. Then I will think of how can I add that field.

If you can give any help/idea that will make my edit process faster, I would be grateful.

**Reply by glenn-jocher** [reply]
@safaATcurrus you need to many changes in multiple places, i.e.:

- model architecture (output layers)
- dataloader
- loss function
- validation metrics (optional)
- plotting/visualization functions (optional?)
